python/cudf/cudf/core/udf/utils.py

```diff
@@ -308,7 +310,7 @@ def _compile_or_get(frame, func, args, kernel_getter=None):
         raise TypeError("only scalar valued args are supported by apply")
 
     # check to see if we already compiled this function
-    cache_key = _generate_cache_key(frame, func, args)
+    cache_key = _generate_cache_key(frame, func, args, suffix=suffix)
```

Comment:
Is it important for us to distinguish between functions compiled for groupby-apply vs normal apply?

Reply:
Yes, it is. The reason is that the cache key is generated based on the dtypes of the input frame and the function itself. It's possible you could get a cache hit for a groupby function where you wanted an apply function, which I am guessing would cause a segfault. So this parameter breaks any degeneracies.